File: backend/nightbiteApi/src/main/java/ni/edu/uam/nightbiteapi/dto/UserRegisterRequest.java
package ni.edu.uam.nightbiteapi.dto;

public class UserRegisterRequest {

    private String username;
    private String email;
    private String password;
    private Integer age;

    public UserRegisterRequest() {
    }

    public UserRegisterRequest(String username, String email, String password, Integer age) {
        this.username = username;
        this.email = email;
        this.password = password;
        this.age = age;
    }

    public String getUsername() {
        return username;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public Integer getAge() {
        return age;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public void setAge(Integer age) {
        this.age = age;
    }
}

File: mobile/nightbiteApp/app/src/main/java/ni/edu/uam/nightbiteapp/data/remote/dto/UserRegisterRequest.kt
package ni.edu.uam.nightbiteapp.data.remote.dto

/**
 * DTO utilizado para enviar los datos de registro de una cuenta de usuario.
 *
 * Representa la cuenta real del jugador, no el personaje dentro del juego.
 */
data class UserRegisterRequest(
    val username: String,
    val email: String,
    val password: String,
    val age: Int
)